test/pthread/test_pthread_proxying_canceled_work.c

```diff
@@ -0,0 +1,83 @@
+#include <assert.h>
+#include <emscripten/proxying.h>
+#include <pthread.h>
+#include <stdio.h>
+
+em_proxying_queue* queue;
+
+void explode(void* arg) { assert(0 && "the work should not be run!"); }
+
+void set_flag(void* flag) {
+  // Schedule the flag to be set on the next turn of the event loop so that we
+  // can be sure cleanup has finished first.
+
+  // clang-format off
+  EM_ASM({setTimeout(() => Atomics.store(HEAP32, $0 >> 2, 1))}, flag);
```

Comment:
Does this need to be `EM_ASM` or can you just use `emscripten_set_timeout`?

Reply:
It needs to be an `EM_ASM` because we need to avoid `callUserCallback`. This code will run after the thread runtime has exited.

Reply:
I see, maybe a comment then?